txs/handler_test.go

```diff
@@ -37,8 +37,9 @@ func Test_WrongHash(t *testing.T) {
 	require.ErrorIs(t, err, errWrongHash)
 	require.ErrorIs(t, err, pubsub.ErrValidationReject)
 	cstate.EXPECT().GetMeshTransaction(tx.ID).Return(nil, nil)
-	req := smocks.NewMockValidationRequest(ctrl)
-	req.EXPECT().Parse().Times(1).Return(tx.TxHeader, nil)
+	req := smocks.NewMockValidationRequestNew(ctrl)
+	req.EXPECT().Cache().Times(1).Return(nil)
```

Comment:
NIT: `Times(1)` and `Return(nil)` are defaults, so it's equivalent to:
```suggestion
	req.EXPECT().Cache()
```

IMHO less is more and it's more readable (less cluttered) this way.